engine/src/test/java/org/hibernate/search/engine/cfg/spi/ConfigurationPropertyInvalidSimpleValuesTest.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.cfg.spi;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import org.hibernate.search.engine.cfg.ConfigurationPropertySource;
+import org.hibernate.search.util.SearchException;
+import org.hibernate.search.util.impl.test.SubTest;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+
+@RunWith(Parameterized.class)
+public class ConfigurationPropertyInvalidSimpleValuesTest<T> extends EasyMockSupport {
+
+	@Parameterized.Parameters(name = "{2}")
+	public static Object[][] data() {
+		return new Object[][] {
+				params(
+						KeyContext::asInteger, "foo", 42,
+						"For input string: \"foo\"",
+						InvalidType.class.getName() + " cannot be cast to java.lang.String"
```

Comment:
> Hum, is it me or you're trying to convert a string to an integer and the error message says you cannot cast it to a String?

The parameters passed to "params" are used for different tests. `For input string: \"foo\"` is what you get when a `NumberFormatException` occurs.

The second error message is only used in a specific test where we put an Object in the properties, in which case Search will see it's not a Number, and as a last resort will try to cast it to a String and parse the string as an Integer. Hence the error message.

> BTW, if it's a JVM error message, I strongly advise to test it because they change and it quickly becomes a nightmare (been there, done that).

I suppose you advise *against* testing it? Because I'm already testing it, so...

Reply:
I changed this commit and force-pushed; both the test and the behavior of parsing utils should now be more... sane.

Reply:
Yeah sorry, was intended as "not to test it".

The tests look more like what I expected, thanks.